master/internal/command/shell_manager.go

```diff
@@ -52,12 +60,63 @@ func (s *shellManager) Receive(ctx *actor.Context) error {
 		}
 		ctx.Respond(resp)
 
+	case ShellLaunchRequest:
+		summary, err := s.processShellLaunchRequest(ctx, msg)
+		if err != nil {
+			ctx.Respond(errors.Wrap(err, "failed to launch shell"))
+		} else {
+			ctx.Respond(summary.ID)
+		}
+
 	case echo.Context:
 		s.handleAPIRequest(ctx, msg)
 	}
 	return nil
 }
 
+func (s *shellManager) processShellLaunchRequest(
+	ctx *actor.Context,
+	req ShellLaunchRequest,
+) (*summary, error) {
+	commandReq, err := parseCommandRequest(*req.User, s.db, req.CommandParams,
+		&s.taskSpec.TaskContainerDefaults,
+	)
```

Comment:
Go style nit: pull this paren up or put all the params on line 82, either way. I wish the formatter handled this, it's really inconsistent all over our code and makes it just a bit harder to skim.

Reply:
how about 
```
	commandReq, err := parseCommandRequest(
		*req.User, s.db, req.CommandParams,
		&s.taskSpec.TaskContainerDefaults,
	)
```